### ISSUE TITLE
rustdoc doesn't render generic params & where-clauses of cross-crate GATs in certain sections

### ISSUE THREAD

**Issue by fmease** [rust-lang/rust#112904]
While rustdoc does indeed render generic parameters and where-clauses of cross-crate GATs in the `.item-decl` of traits or in the section *Required Associated Types* for example, it doesn't do so in section *Implementors* on trait pages or section *Trait Implementations* on an implementor's page. I haven't exhaustively checked yet where precisely they're dropped and where they aren't.

Reproducer:

```rs
// a.rs

pub struct S;

pub trait Tr {
    type F<T>
    where
        (T,): Clone;
}

impl Tr for S {
    type F<T> = T
    where
        (T,): Clone;
}
```

```rs
// b.rs

pub use a::*;
```

```sh
rustc a.rs --crate-type=lib --edition=2021
rustdoc b.rs --crate-type=lib --edition=2021 --extern=a=liba.rlib
```

![Screenshot 2023-06-21 at 20-44-55 Tr in unx - Rust](https://github.com/rust-lang/rust/assets/14913065/2d67c45b-b053-4b06-a0f5-1f20d38c7851)

@rustbot claim
@rustbot label C-bug T-rustdoc A-cross-crate-reexports

**Comment by compiler-errors** [comment]
strange that this wasn't fixed by #103190

**Comment by fmease** [comment]
that's true and also came to my mind when I noticed this, looking into it
